yarpcerrors/errors.go

```diff
@@ -86,6 +85,21 @@ func (s *Status) WithName(name string) *Status {
 		code:    s.code,
 		name:    name,
 		message: s.message,
+		details: s.details,
+	}
+}
+
+// WithDetails returns a new Status with the given details.
+func (s *Status) WithDetails(i []interface{}) *Status {
+	if s == nil {
+		return nil
+	}
+
+	return &Status{
+		code:    s.code,
+		name:    s.name,
+		message: s.message,
+		details: i,
```

Comment:
This doesn't seem right? Shouldn't this be an append sort of situation?

Reply:
Also, while we're at it, WithDetails should probably take a varargs of `...interface{}`?